src/main/scala/com/databricks/labs/overwatch/pipeline/GoldTransforms.scala

```diff
@@ -402,8 +403,21 @@ trait GoldTransforms extends SparkSessionWrapper {
       (driverComputeCost + driverDBUCost + workerComputeCost + workerDBUCost).alias("total_cost")
     )
 
-    clusterPotential
+    val clsfDF = clusterPotential
       .select(clusterStateFactCols: _*)
+
+    val clsfDF1 =clsfDF.filter('state === "TERMINATING").filter('isAutomated === true).filter('days_in_state > 2)
```

Comment:
This handles one of the corner cases, but doesn't help us in cases where the job finishes and we're missing the TERMINATING event altogether. 
In my opinion, we should approach it from the the other side, find the job runs that used an automated cluster and terminated, then find those on the list for clsf and mark them as terminated